### PR TITLE
Restored recob::Trajectory I/O rule lost long time ago.

### PR DESCRIPTION
**Pull request by PetrilloAtWork**
It should fix an issue reported by Kirsty Duffy on reading old (MicroBooNE production) files.

### CODE REVIEW

File: lardataobj/RecoBase/TrackingDicts/classes_def.xml [hunk 1]
```diff
@@ -43,6 +43,39 @@
...
+        <!-- * positions -->
+  <ioread 
+    version="[-10]" 
+    sourceClass="recob::Trajectory"
+    source="std::vector<ROOT::Math::PositionVector3D<ROOT::Math::Cartesian3D<double>, ROOT::Math::GlobalCoordinateSystemTag>> fPositions;"
+    targetClass="recob::Trajectory"
+    target="fPositions"
+    include="Math/GenVector/CoordinateSystemTags.h;Math/GenVector/Cartesian3D.h;Math/GenVector/PositionVector3D.h">
+    <![CDATA[
+      fPositions.clear();
```
Comment (knoepfel) [inline lardataobj/RecoBase/TrackingDicts/classes_def.xml:57]: Calling clear is not necessary, both here and below.

Comment (PetrilloAtWork) [reply lardataobj/RecoBase/TrackingDicts/classes_def.xml:57]: Probably. I did not check the context this snipped is inserted too: I can imagine `fPositions` is a local variable just default-constructed, but I went for the safest approach nevertheless.
After all, I am effectively setting a private data member from outside the class, so I don't want to assume standard rules apply.

Comment (knoepfel) [reply lardataobj/RecoBase/TrackingDicts/classes_def.xml:57]: After talking with Philippe, it turns out that the calling clear is the safest thing to do.  So I withdraw my comments regarding it.

**Comment by knoepfel** [comment]
Assuming L2 approval, this can go in.

**Comment by lgarren** [comment]
trigger build

**Comment by absolution1** [comment]
I will give L2 but I guess we'll need to re-trigger the CI after the experiments have updated their product_deps

**Comment by absolution1** [comment]
approve

**Comment by knoepfel** [comment]
approve

**Comment by knoepfel** [comment]
trigger build

**Pull request merged**